src/HDF5.jl

```diff
@@ -1,5 +1,9 @@
 module HDF5
 
+if isdefined(Base, :Experimental) && isdefined(Base.Experimental, Symbol("@optlevel"))
```

Comment:
Perhaps we could save this for another PR ?

Reply:
The only reason I say this is that it is not directly related to the rest of the changes in the PR. Perhaps it also should have some further discussion.

Reply:
Done - #682